pype/modules/ftrack/actions/action_delivery.py

```diff
@@ -381,21 +387,36 @@ def launch(self, session, entities, event):
 
         try:
             self.db_con.install()
-            self.real_launch(session, entities, event)
-            job["status"] = "done"
+            result = self.real_launch(session, entities, event)
+            if result["success"]:
+                job["status"] = "done"
+            else:
+                job["status"] = "failed"
 
-        except Exception:
+        except Exception as exc:
+            job["status"] = "failed"
+            result = {
```

Comment:
Rather use `self.show_interface(items, "Delivery failed", event)` to show the label. When items are returned as result of this action and user would click on `"Submit"` button it would lead back to action `_launch` method which is not expected.

Reply:
Didn't know about 'show_interface' method.
Much better than having the callback on submit button :)